packages/components/src/components/brandshape/brandshape.ts

```diff
@@ -0,0 +1,155 @@
+import { createRef, ref } from 'lit/directives/ref.js';
+import { css } from 'lit';
+import { customElement, property, state } from 'lit/decorators.js';
+import { debounce } from 'throttle-debounce';
+import { html } from 'lit/static-html.js';
+import componentStyles from '../../styles/component.styles';
+import cx from 'classix';
+import SolidElement from '../../internal/solid-element';
+import type { Ref } from 'lit/directives/ref.js';
+import type { TemplateResult } from 'lit-html';
+
+/**
+ * @summary The Brandshape highlights a piece of content.
+ * @documentation https://solid.union-investment.com/[storybook-brandshape]/brandshape
+ * @status
+ * @since
+ *
+ * @slot - The content inside the brandshape.
+ *
+ * @csspart base - The component's base wrapper.
+ * @csspart content-center - Center content wrapper.
+ * @csspart content-top - Top content wrapper.
+ *
+ * @cssproperty --brandshape-opacity - The opacity of the brandshape.
+ */
+@customElement('sd-brandshape')
+export default class SdBrandshape extends SolidElement {
+  /** The brandshape's theme variant. */
+  @property({ type: String }) variant: 'neutral-100' | 'primary' | 'white' = 'primary';
+
+  /** Defines the form of the brandshape  */
+  @property({ type: String }) form: 'full' | 'top' | 'topCenter' | 'centerBottom' = 'full';
+
+  @state() private componentBreakpoint: 0 | 560 = 0;
+
+  private resizeObserver: ResizeObserver;
+
+  private containerRef: Ref<HTMLInputElement> = createRef();
+
+  private get topSvg(): TemplateResult {
+    return html`<svg xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 958 166">
+      <path
+        class=${this.svgClasses()}
+        d="M836.828 1.894.001 164.785V168h958v-66.312c0-6.524-.628-13.033-1.876-19.436-10.74-55.114-64.151-91.092-119.297-80.358Z"
+      />
+    </svg>`;
+  }
+
+  private get bottomSvg(): TemplateResult {
+    return html`<svg xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 958 168">
+      <path
+        class=${this.svgClasses()}
+        d="M121.173 166.106 958 3.216V0H0v66.312c0 6.524.628 13.033 1.876 19.436 10.74 55.114 64.151 91.092 119.297 80.358Z"
+      />
+    </svg>`;
+  }
+
+  private svgClasses(): string {
+    return cx(
+      {
+        'neutral-100': 'fill-neutral-100/[var(--brandshape-opacity)]',
+        primary: 'fill-primary/[var(--brandshape-opacity)]',
+        white: 'fill-white/[var(--brandshape-opacity)]'
```

Comment:
Can you use all transparent variants from a brand perspective?

Still, I believe we need either more variants e. . `primary-opaque`, `neutral-100-opaque` or have a boolean `opaque' if this really is for every bg-color.

Background: Brand and Design system define the allowed opacity (AFAIK), not the user of the component.

Reply:
In the cms-context we need to be able to set the following opacities : 80%, 90%, 100% on white and primary. We could also make this more strict so only this 3 options are available

Reply:
Okay, this should be definitely more strict, that doesn't sound very systematic.

Maybeeeee... as a solution for the moment you could just set the color and override the opacity with the part selector and pointing to the SVG?

I believe we will introduce opacity later in some kind of property, but maybe for the moment to get you running in your project this would be easier?